frontend/SplitPane.js

```diff
@@ -39,30 +44,45 @@ class SplitPane extends React.Component {
     super(props);
     this.state = {
       moving: false,
-      width: props.initialWidth,
+      width: (props.isVertical) ? '100%' : props.initialWidth,
+      height: (!props.isVertical) ? '100%' : props.initialHeight,
     };
   }
 
-  onMove(x: number) {
+  componentWillReceiveProps(nextProps: Props) {
+    this.setState({
```

Comment:
Looks like there's no need to run this code if `nextProps.isVertical` hasn't changed.

```js
if (nextProps.isVertical !== this.props.isVertical) {
  // put your code here
}
```